plugins/modules/zpool.py

```diff
@@ -0,0 +1,588 @@
+                vdevs.append({'name': base_device, 'type': 'stripe', 'disks': [base_device]})
+        if current:
+            vdevs.append(current)
+        return vdevs
+
+    def remove_vdevs(self):
+        current = self.list_vdevs_with_names()
+        current_disks = {disk for vdev in current for disk in vdev['disks']}
+        desired_disks = {disk for vdev in self.vdevs for disk in vdev.get('disks', [])}
+        gone = current_disks - desired_disks
+        to_remove = [vdev['name'] for vdev in current if any(disk in gone for disk in vdev['disks'])]
+        if not to_remove:
+            return {}
+        cmd = [self.zpool_cmd, 'remove']
+        if self.module.check_mode:
+            cmd.append('-n')
+        cmd.append(self.name)
+        cmd.extend(to_remove)
+        rc, stdout, stderr = self.module.run_command(cmd, check_rc=True)
+        self.changed = True
+        if self.module.check_mode:
+            return {'prepared': stdout}
+        before = [vdev['name'] for vdev in current]
+        after = [name for name in before if name not in to_remove]
+        return {'before': {'vdevs': before}, 'after': {'vdevs': after}}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            state=dict(type='str', choices=['present', 'absent'], default='present'),
+            disable_new_features=dict(type='bool', default=False),
+            force=dict(type='bool', default=False),
+            pool_properties=dict(type='dict', default={}),
+            filesystem_properties=dict(type='dict', default={}),
+            mountpoint=dict(type='str', required=False),
+            altroot=dict(type='str', required=False),
+            temp_name=dict(type='str', required=False),
+            vdevs=dict(type='list', required=True, elements='dict'),
+        ),
+        supports_check_mode=True
+    )
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+    disable_new_features = module.params.get('disable_new_features')
+    force = module.params.get('force')
+    pool_properties = module.params.get('pool_properties')
+    filesystem_properties = module.params.get('filesystem_properties')
+    mountpoint = module.params.get('mountpoint')
+    altroot = module.params.get('altroot')
+    temp_name = module.params.get('temp_name')
+    vdevs = module.params.get('vdevs')
+
+    for property_key in ('pool_properties', 'filesystem_properties'):
+        for key, value in list(module.params.get(property_key, {}).items()):
+            if isinstance(value, bool):
+                module.params[property_key][key] = 'on' if value else 'off'
+
+    for idx, vdev in enumerate(vdevs, start=1):
+        disks = vdev.get('disks')
+        if not isinstance(disks, list) or len(disks) == 0:
+            module.fail_json(msg="vdev #{idx}: at least one disk is required (got: {disks!r})".format(idx=idx, disks=disks))
```

Comment:
This validation can be performed automatically by Ansible if you declare the suboptions in `vdevs` (line 484 above).